### ISSUE TITLE
Edge node is unable to update node status after network disconnection

### ISSUE THREAD

**Issue by federicocagnola** [kubeedge/kubeedge#4850]
**What happened**:
Edge node network disconnection stops edgecore from reconnecting to cloud. Sync message keeps failing after disconnection.
```
lug 14 10:16:40 v016-lb edgecore[649]: I0714 10:16:40.664493     649 edgedmetricsconnection.go:119] receive stop signal, so stop metrics scan ...
lug 14 10:16:45 v016-lb edgecore[649]: I0714 10:16:45.676794     649 edgedmetricsconnection.go:119] receive stop signal, so stop metrics scan ...
lug 14 10:16:52 v016-lb edgecore[649]: I0714 10:16:52.720844     649 edgedmetricsconnection.go:119] receive stop signal, so stop metrics scan ...
lug 14 10:17:40 v016-lb edgecore[649]: I0714 10:17:40.683680     649 edgedmetricsconnection.go:119] receive stop signal, so stop metrics scan ...
lug 14 10:18:11 v016-lb edgecore[649]: E0714 10:18:11.037024     649 reader.go:33] failed to read package header from buffer
lug 14 10:18:11 v016-lb edgecore[649]: E0714 10:18:11.037049     649 quic.go:223] failed to read message, error: NetworkIdleTimeout: No recent network activity.
lug 14 10:18:17 v016-lb edgecore[649]: I0714 10:18:17.189062     649 edgedmetricsconnection.go:85] receive remove client id 1033351
lug 14 10:18:17 v016-lb edgecore[649]: I0714 10:18:17.189088     649 edgedmetricsconnection.go:119] receive stop signal, so stop metrics scan ...
lug 14 10:18:17 v016-lb edgecore[649]: I0714 10:18:17.193518     649 edgedmetricsconnection.go:119] receive stop signal, so stop metrics scan ...
lug 14 10:18:19 v016-lb edgecore[649]: E0714 10:18:19.038871     649 session.go:27] failed to open stream, error: NetworkIdleTimeout: No recent network activity.
lug 14 10:18:19 v016-lb edgecore[649]: E0714 10:18:19.038920     649 quic.go:161] failed to open stream, error: NetworkIdleTimeout: No recent network activity.
lug 14 10:18:19 v016-lb edgecore[649]: W0714 10:18:19.038926     649 smgr.go:313] get stream error(NetworkIdleTimeout: No recent network activity.)
lug 14 10:18:19 v016-lb edgecore[649]: E0714 10:18:19.038931     649 quic.go:306] failed to acquire stream sync, error:NetworkIdleTimeout: No recent network activity.
lug 14 10:18:19 v016-lb edgecore[649]: E0714 10:18:19.038940     649 process.go:183] websocket write error: failed to send message, error: failed to acquire stream sync, error:NetworkIdleTimeout: No recent network a>
lug 14 10:18:19 v016-lb edgecore[649]: W0714 10:18:19.038982     649 edgehub.go:126] connection is broken, will reconnect after 30s
lug 14 10:18:19 v016-lb edgecore[649]: I0714 10:18:19.038994     649 process.go:299] DeviceTwin receive msg
lug 14 10:18:19 v016-lb edgecore[649]: W0714 10:18:19.039013     649 eventbus.go:168] Action not found
lug 14 10:18:19 v016-lb edgecore[649]: I0714 10:18:19.039022     649 process.go:68] Send msg to the CommModule module in twin
lug 14 10:18:19 v016-lb edgecore[649]: I0714 10:18:19.038994     649 process.go:461] node connection event occur: cloud_disconnected
lug 14 10:18:19 v016-lb edgecore[649]: I0714 10:18:19.039046     649 process.go:461] node connection event occur: cloud_disconnected
lug 14 10:18:29 v016-lb edgecore[649]: I0714 10:18:29.864980     649 edgedmetricsconnection.go:119] receive stop signal, so stop metrics scan ...
lug 14 10:18:40 v016-lb edgecore[649]: I0714 10:18:40.672694     649 edgedmetricsconnection.go:119] receive stop signal, so stop metrics scan ...
lug 14 10:18:45 v016-lb edgecore[649]: I0714 10:18:45.690008     649 edgedmetricsconnection.go:119] receive stop signal, so stop metrics scan ...
lug 14 10:18:47 v016-lb edgecore[649]: E0714 10:18:47.819935     649 metaclient.go:117] send sync message kube-node-lease/lease/mt01-lb failed, error:timeout to get response for message 91ac6d67-1705-4cbf-8d61-d805e>
lug 14 10:18:47 v016-lb edgecore[649]: E0714 10:18:47.825967     649 session.go:27] failed to open stream, error: NetworkIdleTimeout: No recent network activity.
lug 14 10:18:47 v016-lb edgecore[649]: E0714 10:18:47.825983     649 quic.go:161] failed to open stream, error: NetworkIdleTimeout: No recent network activity.
lug 14 10:18:47 v016-lb edgecore[649]: W0714 10:18:47.825989     649 smgr.go:313] get stream error(NetworkIdleTimeout: No recent network activity.)
lug 14 10:18:47 v016-lb edgecore[649]: E0714 10:18:47.825992     649 quic.go:306] failed to acquire stream sync, error:NetworkIdleTimeout: No recent network activity.
lug 14 10:18:47 v016-lb edgecore[649]: E0714 10:18:47.826001     649 process.go:161] failed to send message to cloud: failed to send message, error: failed to acquire stream sync, error:NetworkIdleTimeout: No recent>
lug 14 10:18:49 v016-lb edgecore[649]: I0714 10:18:49.040034     649 quicclient.go:45] Quic start to connect Access
lug 14 10:18:49 v016-lb edgecore[649]: I0714 10:18:49.237588     649 quic.go:89] get response: {Header:{ID:48262572-ef85-4982-9f7a-b3e96bdd1e53 ParentID:5d5cfcb5-03a2-4a8b-a197-d782cd3e38ee Timestamp:1689322729208 R>
lug 14 10:18:49 v016-lb edgecore[649]: I0714 10:18:49.237623     649 quic.go:115] connect remote peer successfully
lug 14 10:18:49 v016-lb edgecore[649]: I0714 10:18:49.237639     649 quicclient.go:81] Quic connect to cloud access successful
lug 14 10:18:49 v016-lb edgecore[649]: I0714 10:18:49.237683     649 process.go:461] node connection event occur: cloud_connected
lug 14 10:18:49 v016-lb edgecore[649]: W0714 10:18:49.237699     649 eventbus.go:168] Action not found
lug 14 10:18:49 v016-lb edgecore[649]: I0714 10:18:49.237716     649 process.go:461] node connection event occur: cloud_connected
lug 14 10:18:49 v016-lb edgecore[649]: I0714 10:18:49.237733     649 process.go:299] DeviceTwin receive msg
lug 14 10:18:49 v016-lb edgecore[649]: I0714 10:18:49.237746     649 process.go:68] Send msg to the CommModule module in twin
lug 14 10:18:50 v016-lb edgecore[649]: E0714 10:18:50.681211     649 metaclient.go:117] send sync message default/node/mt01-lb failed, error:timeout to get response for message c69fe684-bc55-4347-9b53-477ca53acec6, >
lug 14 10:18:50 v016-lb edgecore[649]: E0714 10:18:50.681215     649 process.go:422] remote query failed, req[msgID[8caa5b4b-7440-4e35-bf1e-3e658c959455] resource[default/node/mt01-lb]], err: timeout to get response>
lug 14 10:18:50 v016-lb edgecore[649]: W0714 10:18:50.681316     649 context_channel.go:164] Get bad anonName: when sendresp message, do nothing
```
At 10:18:19 `edgecore` logged `node connection event occur: cloud_disconnected`. Thirty seconds later edgecore marked the status as reconnected at 10:18:49: `node connection event occur: cloud_connected`. Nonetheless after this point the connection was broken, node status was not updated and any deploy/remove/install action on pods failed because the node was marked as `NetworkUnavailable` and not `Ready`. Following this issue edgecore kept logging node update failures like: `send sync message kube-node-lease/lease/mt01-lb failed, error:timeout to get response for message`

**What you expected to happen**:
Edge node reconnection should cause a node sync.

**How to reproduce it (as minimally and precisely as possible)**:
Disconnect edge node and follow logs.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.24.6
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`): v1.12.2
- <details><summary>Cloud nodes Environment:</summary>

  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>
- <details><summary>Edge nodes Environment:</summary>

  - edgecore version (e.g. `edgecore --version`): v1.12.2
  - Hardware configuration (e.g. `lscpu`): `x86_64` `Intel(R) Core(TM) i5-1035G1 CPU @ 1.00GHz`
  - OS (e.g. `cat /etc/os-release`): `Ubuntu 20.04 LTS (Linux)`
  - Kernel (e.g. `uname -a`): `5.15.0-76-generic`
  - Go version (e.g. `go version`): N/A
  - Others: N/A

  </details>

**Comment by Shelley-BaoYue** [comment]
Do you mean that the edgenode is still `not ready` after reconnection? Have you ever waited for a longer period of time?  because some accumulated data needs to be processed, it may take some time.

**Comment by federicocagnola** [comment]
@Shelley-BaoYue I did wait, the problem is precisely this, it _never reconnects_ and keeps logging this:
```
send sync message kube-node-lease/lease/mt01-lb failed, error:timeout to get response for message
```
this persists until reboot / manual restart with `systemctl restart edgecore.service`

**Comment by Shelley-BaoYue** [comment]
Sorry, I failed to reproduce it in my environment.  Could you please provide more info including the yaml config of cloudcore and edgecore?

**Comment by Shelley-BaoYue** [comment]
Is this a inevitable issue? It would be nice to provide cloudcore logs, especially the logs after reconnection.

**Comment by federicocagnola** [comment]
fixed from v-1.14.0

**Comment by GuhAlex** [comment]
please , update the documentation here https://release-1-20.docs.kubeedge.io/docs/setup/install-with-binary
